chipsec/logger.py

```diff
@@ -208,10 +208,17 @@ def flush(self):
     def set_always_flush( self, val ):
         self.ALWAYS_FLUSH = val
 
-    def log( self, text, level=pyLogging.INFO):
-        """Sends plain text to logging."""
+    def add_text_to_results(self, text):
         if self.Results.get_current() is not None:
+            if hasattr(text, '__getitem__'):
+                text = "".join(text)
+            elif not text.endswith('\n'):
+                text += ('\n')
             self.Results.get_current().add_output(text)
+
+    def log( self, text, level=pyLogging.INFO):
+        """Sends plain text to logging."""
+        self.add_text_to_results(text)
```

Comment:
It looks like log calls add_text_to_results, why is this added to each logging function that also calls log?

Reply:
The idea was to have whatever is being logged by the tests show up as well in the results. Currently if you export the results to xml, json (or the proposed markdown) you don't get to see in the file what is the test complaining about when it is failing.

Reply:
I am saying the calls on line 233, 277, 283, etc are redundant as each function calls log where your function is called.  I think that it can be condensed to only calling it within the log function

Reply:
I see what you mean. The way I can see we can simplify this change would be to make a new class deriving from the logging module and then make all the callers use this new class.

It is cleaner and easier to maintain but a little more invasive. I can work on that idea if that sounds good to you.